src/cpp/core/FileLock.cpp

```diff
@@ -15,163 +15,242 @@
 
 #include <core/FileLock.hpp>
```

Comment:
Let's rename this source file to FileLockCore.cpp so that the diff shows it as a brand new file (easier to read and comment)